templates/ec2/sc-ec2-linux-docker-notebook.yaml

```diff
@@ -0,0 +1,491 @@
+    Metadata:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E1022
+            - W1011
+      'AWS::CloudFormation::Init':
+        configSets:
+          StartContainersJupyter:
+            - start_docker_network
+            - start_reverse_proxy
+            - make_work_dir
+            - start_jupyter_notebook
+          StartContainersRstudio:
+            - start_docker_network
+            - start_reverse_proxy
+            - make_work_dir
+            - start_rstudio_notebook
+        start_docker_network:
+          commands:
+            start_docker_network:
+              command: >
+                docker network inspect ${NETWORK_NAME} ||
+                docker network create --driver bridge ${NETWORK_NAME}
+              env:
+                NETWORK_NAME:
+                  'Fn::FindInMap': [GlobalVars, DockerNetworkName, Value]
+        start_reverse_proxy:
+          commands:
+            start_reverse_proxy:
+              # For idempotency this command tries to restart an existing container and only creates one if that fails.
+              command: >
+                docker start reverse-proxy ||
+                docker run -d -p 443:443 --memory=1g --cpus=1
+                --name reverse-proxy
+                --network ${NETWORK_NAME}
+                --restart unless-stopped
+                -e EC2_INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+                -e NOTEBOOK_HOST=${NOTEBOOK_CONTAINER_NAME}
+                -e AWS_REGION=${AWS_REGION}
+                -e SERVICE_CATALOG_PREFIX=${SERVICE_CATALOG_PREFIX}
+                -e SSM_PARAMETER_SUFFIX=${SSM_PARAMETER_SUFFIX}
+                ghcr.io/Sage-Bionetworks-IT/notebook-reverse-proxy-${NOTEBOOK_TYPE}:v1.0.0
+              env:
+                AWS_REGION: !Ref AWS::Region
+                NETWORK_NAME:
+                  'Fn::FindInMap': [GlobalVars, DockerNetworkName, Value]
+                NOTEBOOK_CONTAINER_NAME:
+                  'Fn::FindInMap': [GlobalVars, NotebookContainerName, Value]
+                SERVICE_CATALOG_PREFIX:
+                  'Fn::FindInMap': [GlobalVars, SSMParameterPrefix, Value]
+                SSM_PARAMETER_SUFFIX:
+                  'Fn::FindInMap': [GlobalVars, SSMParameterSuffix, Value]
+                NOTEBOOK_TYPE:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, NoteBookTypeForDocker]
+        make_work_dir:
+          commands:
+            make_work_dir:
+              command: |
+                mkdir -p ${EC2_WORK_DIR}
+                sudo chmod 777 ${EC2_WORK_DIR}
+              env:
+                EC2_WORK_DIR:
+                  'Fn::FindInMap': [NotebookTypeToInitParams, !Ref NotebookType, EC2WorkDir]
+        start_jupyter_notebook:
+          commands:
+            start_notebook:
+              command: |
+                EC2_INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
```

Comment:
shouldn't this env var be passed into the docker container (with `-e` flag) as well?

Reply:
I don't think so: From what I can tell the values to be passed into the container are `SYNAPSE_TOKEN_AWS_SSM_PARAMETER_NAME` and `AWS_DEFAULT_REGION`. (See [this](https://raw.githubusercontent.com/Sage-Bionetworks/service-catalog-utils/v2.0.3/linux/opt/sage/bin/rstudio_synapse_cred_env.sh) script.) The EC2 instance ID is embedded in `SYNAPSE_TOKEN_AWS_SSM_PARAMETER_NAME` but not otherwise used in the Notebook.  If you feel I am mistaken please let me know.